### PR TITLE
Add integration tests to Conseil

### PR DESCRIPTION
**Pull request by PrathamRawat**
Develop integration tests for Conseil

This runs a script to test different (Tezos for now) endpoints on Conseil to ensure that everything is working as is known. It does this in Travis, building Conseil from source, and running the API and Lorre for a fixed length of time, before sending requests to the node. This should ensure that in any fix/update, no existing functionality in broken.

If some core behavior has changed, it is expected that the script's Constants should be updated to ensure that the script is up to date for future versions.

### CODE REVIEW

File: conseil-api-tests/src/main/scala/tech/cyptonomic/conseil/api/tests/ApiTestRun.scala [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+
+        val queryString: String =
+          """
+            |{
+            |     "field": ["%FIELD%"],
+            |     "predicates": [
+            |         {
+            |             "field": "%FIELD%",
+            |             "operation": "isnull",
+            |             "set": [""],
```
Comment (piotrkosecki) [inline conseil-api-tests/src/main/scala/tech/cyptonomic/conseil/api/tests/ApiTestRun.scala:225]: cosmetics: set can be an empty list `[]` for `isnull` operation.

File: conseil-api-tests/src/main/scala/tech/cyptonomic/conseil/api/tests/ApiTestRun.scala [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+
+  /**
+   * Validate a query output by searching for JSON keys
+   * @param key The json key to search for
+   * @param result The result of a query to search in
+   * @param notNull Whether the key's value should be allowed to be null or not
+   */
+  def validateJsonByKey(key: String, result: String, notNull: Boolean = false): Unit = {
+
+    if(result.compareTo("") == 0) throw new NullPointerException("Query Returned an Empty String")
```
Comment (piotrkosecki) [inline conseil-api-tests/src/main/scala/tech/cyptonomic/conseil/api/tests/ApiTestRun.scala:274]: why not just `result == ""`?

File: conseil-api-tests/src/main/scala/tech/cyptonomic/conseil/api/tests/ApiTestRun.scala [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+      `Content-Type`(MediaType.application.json),
+      Accept(MediaType.application.json),
+      Header("apiKey", CONSEIl_API_KEY)
+    )
+
+    val result: String = client.resource.use { client =>
+      client.expect[String](conseilRequest)
+    }.unsafeRunSync()
+
+    if(validationItems.nonEmpty) validationItems.foreach(item => {
```
Comment (piotrkosecki) [inline conseil-api-tests/src/main/scala/tech/cyptonomic/conseil/api/tests/ApiTestRun.scala:117]: you can write `validationItems.foreach(item => { ` without using `(` - similarly in other cases below

File: conseil-api-tests/src/main/scala/tech/cyptonomic/conseil/api/tests/ApiTestRun.scala [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+   * @param key The json key to search for
+   * @param result The result of a query to search in
+   * @param notNull Whether the key's value should be allowed to be null or not
+   */
+  def validateJsonByKey(key: String, result: String, notNull: Boolean = false): Unit = {
+
+    if(result.compareTo("") == 0) throw new NullPointerException("Query Returned an Empty String")
+
+    parse(result) match {
+      case Left(failure) => throw new IllegalArgumentException("Query did not return valid JSON")
```
Comment (piotrkosecki) [inline conseil-api-tests/src/main/scala/tech/cyptonomic/conseil/api/tests/ApiTestRun.scala:277]: `failure` is not used an can be replaced with `_`

File: conseil-api-tests/src/main/scala/tech/cyptonomic/conseil/api/tests/ApiTestRun.scala [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+            |     "limit": 1
+            |}
+            |""".stripMargin.replace("%FIELD%", attribute)
+
+        println("\n\n\nValidating Tezos " + entity._1 + " " + attribute + " Data\n")
+        var result: String = ""
+        try {
+          result = sendConseilQuery(httpClient, CONSEIL.withPath(Requests.getTezosQueryPath(entity._1)), queryString)
+        } catch {
+          case x: org.http4s.client.UnexpectedStatus => {
```
Comment (piotrkosecki) [inline conseil-api-tests/src/main/scala/tech/cyptonomic/conseil/api/tests/ApiTestRun.scala:240]: `x` is not used an can be replaced with `_`

File: conseil-api-tests/src/main/scala/tech/cyptonomic/conseil/api/tests/ApiTestRun.scala (hunk 2 above)
Comment (PrathamRawat) [reply conseil-api-tests/src/main/scala/tech/cyptonomic/conseil/api/tests/ApiTestRun.scala:274]: Oh I thought like Java, string comparison might only be properly done through comparison methods

File: conseil-api-tests/src/main/scala/tech/cyptonomic/conseil/api/tests/ApiTestRun.scala [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+            |     ],
+            |     "orderBy": [],
+            |     "aggregation": [],
+            |     "limit": 1
+            |}
+            |""".stripMargin.replace("%FIELD%", attribute)
+
+        println("\n\n\nValidating Tezos " + entity._1 + " " + attribute + " Data\n")
+        var result: String = ""
+        try {
```
Comment (piotrkosecki) [inline conseil-api-tests/src/main/scala/tech/cyptonomic/conseil/api/tests/ApiTestRun.scala:237]: I don't like try/catch here and in other cases. I would rather see handling errors with `.handleErrorWith(...)` method on `IO` and making calls asynchronous.

Comment (PrathamRawat) [reply conseil-api-tests/src/main/scala/tech/cyptonomic/conseil/api/tests/ApiTestRun.scala:237]: Why should calls be asynchronous?

Comment (piotrkosecki) [reply conseil-api-tests/src/main/scala/tech/cyptonomic/conseil/api/tests/ApiTestRun.scala:237]: Basically when you are using any Future/IO in Scala it is preferred to compose those effects and use `unsafeRunSync`/`unsafeRunAsync` just once in the main object.

**Comment by vishakh** [comment]
Stale but we should revisit in the future.

**Pull request closed without merging**